app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -92,9 +103,41 @@ class WallpaperManager(private val settings: Settings) {
     private fun getCurrentWallpaperFromSettings(): Wallpaper {
         val currentWallpaper = settings.currentWallpaper
         return if (currentWallpaper.isEmpty()) {
-            Wallpaper.NONE
+            defaultWallpaper
+        } else {
+            availableWallpapers.find { it.name == currentWallpaper } ?: defaultWallpaper
+        }
+    }
+
+    fun loadWallpaperFromAssets(wallpaper: Wallpaper, context: Context): Bitmap {
+        val path = if (isLandscape(context)) {
+            wallpaper.landscapePath
         } else {
-            Wallpaper.valueOf(currentWallpaper)
+            wallpaper.portraitPath
+        }
+        return context.assets.open(path).use {
+            BitmapFactory.decodeStream(it)
         }
     }
+
+    private fun isLandscape(context: Context): Boolean {
+        return context.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
+    }
+
+    fun loadWallpapers() {
+        availableWallpapers.clear()
+        availableWallpapers.add(defaultWallpaper)
+        availableWallpapers.addAll(wallpaperStorage.all())
+        currentWallpaper = getCurrentWallpaperFromSettings()
```

Comment:
I think this `currentWallpaper = getCurrentWallpaperFromSettings()` line can be omitted, as it should be run when the object is instantiated

Reply:
The issue It's not completely straightforward, imagine you have a selected wallpaper that it is one that it is loaded from assets for example `wallpaper_1`, in settings you will have  `settings.currentWallpaper` with the value `wallpaper_1`, when `var currentWallpaper: Wallpaper = getCurrentWallpaperFromSettings()` is initialized we won't be able to find the `wallpaper_1` on `availableWallpapers` and we will fallback to  `defaultWallpaper`, as `availableWallpapers` doesn't have the `wallpaper_1` until we load it from assets.

```kotlin
    val availableWallpapers = mutableListOf(defaultWallpaper)

    var currentWallpaper: Wallpaper = getCurrentWallpaperFromSettings()
        set(value) {
            settings.currentWallpaper = value.name
            field = value
        }

    private fun getCurrentWallpaperFromSettings(): Wallpaper {
        val currentWallpaper = settings.currentWallpaper
        return if (currentWallpaper.isEmpty()) {
            defaultWallpaper
        } else {
            availableWallpapers.find { it.name == currentWallpaper } ?: defaultWallpaper
        }
    }
```

To make sure the `currentWallpaper` is in sync with what we have in `settings.currentWallpaper` we will need to init `currentWallpaper` after we load all the wallpapers from assets.

Reply:
Those things being out of sync seems like it could be a cause of concern. I wonder if we should change the `WallpaperManager::loadWallpapers` function to run on initialization, instead of in `FenixApplication::initializeWallpapers`. Then, the two properties should be in sync. 

My concern there would be a perf regression on app startup. I think we would need to revisit this in a followup to make the wallpaper loading asynchronous.

Reply:
They will be in sync the first time when we load, but if we do subsequent loads like after editing the `wallpapers.json` file it will be out of sync again. In this case, I think there is no way to avoid the performance hit as even moving the load to the initialization we will hit load when we try to access the `wallpaperManager`  normally on the home screen.

Reply:
At least it could help to reduce the time when users are starting the app without entering the home screen

Reply:
I updated the PR